project/index.mjs

```diff
@@ -433,6 +433,34 @@ router.post('/:projectId/removeCustomRoles', auth0Middleware(), async (req, res)
     respondWithError(res, error.status ?? 500, error.message ?? 'Error removing custom roles.')
   }
 })
+ 
+// Update Project Metadata
+router.route("/:projectId/update-metadata").put(auth0Middleware(), async (req, res) => {
+  const { projectId } = req.params
+  const metadata = req.body.metadata
```

Comment:
We have (or will have) the ability to 'set' any unprotected property on the Project, so having to put this body inside a metadata field is extra unless there is additional data that will be needed.

To be clear, the pain point for me is that a dev thinks "I want to update the metadata". RESTfully, the dev PUTs the new value to "./update-metadata" and then also puts the content inside of a `metadata` property instead of just sending a JSON Array. A dev may feel most efficient for the path to just be "I want to update (PUT) this project's metadata (projectId/metadata) with this new content ( `[{},{},{}]` ).

Reply:
We will always find ourselves here, it is a RESTful philosophical `undefined`.  Either way makes logical sense and there is not much guidance beyond that.  

Since we have the fixed `metadata` vocabulary we should not even entertain the possibility of another metadata-like `key`.  From a minimalistic-dodge-inherited-failure-cases point of view the body should be an Array.  Only indexes that are JSON objects will be considered and others are ignored.  The considered objects themselves must contain a `label` and `value` property and other properties are ignored.  Considered objects without both `label` and `value` are ignored.